### DISCUSSION TITLE
nextval('directus_fields_id_seq'::regclass)

### DISCUSSION THREAD

**Discussion by ddcollins** [directus/directus#4437] [Archive]
I've built some scripts around my Directus project to "snapshot" the database tables to json files, so that we may commit them to git and version control the data models.

I have scripts to "load" the snapshots back into the database, so that we can do staging and production deployments off the locally generated "snapshots".

I'm currently snapshotting the `directus_fields`, `directus_collections`, `directus_relations` and any data model tables I created in Directus (non prefixed `directus_` tables). Going to throw out the revisions, activity, etc to reduce complexity.

The problem I am running into is when I "load" these snapshots into the database, then try to go to the Directus interface and create a new data model or record, I get a "duplicate primary key" on the `directus_fields` table.

I presume this is because the Directus auto incrementing id field is set to be a value that is already used in the `directus_fields` table.

Could you provide any information on how the `directus_fields` default `id` column value of `nextval('directus_fields_id_seq'::regclass)` works and if it would be possible for me to somehow get/set this value in Directus to try and work around my problem?

![Image 2021-03-08 at 5 14 40 PM](https://user-images.githubusercontent.com/38921476/110394289-cb730e00-8031-11eb-9d04-2370dc358b09.png)

**Comment by rijkvanzanten** [comment]
You can't get/set the last postgres sequence value through Directus. You'll have to write a bit of extra magic around your snapshotting in order to reset the sequence value to match the injected rows from your snapshot.

In order to reset the max sequence value, you'd want to run something like:

```sql
SELECT SETVAL('public.directus_activity_id_seq', COALESCE(MAX(id), 1) ) FROM public.directus_activity;
```

(where `public.directus_activity_id_seq` is replaced with `<schema>.<table>_id_seq`).

I've used this snippet before to generate the above queries necessary to reset the sequence values to the correct maximums for all your tables:

```sql
SELECT 'SELECT SETVAL(' ||
       quote_literal(quote_ident(PGT.schemaname) || '.' || quote_ident(S.relname)) ||
       ', COALESCE(MAX(' ||quote_ident(C.attname)|| '), 1) ) FROM ' ||
       quote_ident(PGT.schemaname)|| '.'||quote_ident(T.relname)|| ';'
FROM pg_class AS S,
     pg_depend AS D,
     pg_class AS T,
     pg_attribute AS C,
     pg_tables AS PGT
WHERE S.relkind = 'S'
    AND S.oid = D.objid
    AND D.refobjid = T.oid
    AND D.refobjid = C.attrelid
    AND D.refobjsubid = C.attnum
    AND T.relname = PGT.tablename
ORDER BY S.relname;
```

(Also, hopefully this is a stop-gap solution until we have https://github.com/directus/directus/issues/2928 and/or https://github.com/directus/directus/discussions/3891. I'd love to hear your thoughts and use case on there so we can make sure to check all the boxes 👍🏻 )

**Reply by ddcollins** [reply]
This was exactly what I needed, thank you! I'll put some thoughts there once I wrap up development on this and have a complete picture of our use case.
